### PR TITLE
ICMSLST-1663 Add FA-SIL chief licence data edifact converter.

### PR DESCRIPTION
**Pull request by MattHolmes123**
Other changes:
  - Rename test licence references.

### CODE REVIEW

**Review by mikeylockwood — APPROVED** [review]

File: mail/libraries/lite_to_edifact_converter.py [hunk 1]
```diff
@@ -307,3 +307,29 @@ def get_restrictions(licence: LicencePayload) -> chieftypes.Restrictions:
...
+
+    if licence_type == LicenceTypeEnum.IMPORT_SIL:
+        # each line can have "controlled_by": "Q" or "O"
+        for idx, good in goods_iter:
+            controlled_by = good["controlled_by"]
+
+            if controlled_by == "Q":
+                kwargs = {"quantity_unit": "030", "quantity_issued": good["quantity"]}
+            else:
+                kwargs = {}
```
[the remark this answers is not in the document]
Comment (saruniitr) [reply mail/libraries/lite_to_edifact_converter.py:326]: Could you also create constants for `Q`, `O` and the quantity code `030` as there are multiple usages? not necessarily in this PR.

File: mail/tests/test_serializers.py [hunk 2]
```diff
@@ -373,3 +373,63 @@ def test_valid_fa_dfl_payload(self):
...
+    return {
+        "type": "SIL",
+        "action": "insert",
+        "id": str(uuid.uuid4()),
+        "reference": "GBSIL3333333H",
+        "case_reference": "IMA/2022/00003",
+        "start_date": "2022-06-29",
+        "end_date": "2024-12-29",
+        "organisation": {
+            "eori_number": "123456654321",
```
Comment (saruniitr) [inline mail/tests/test_serializers.py:421]: I think in one of the serializer there are constraints setup for EORI number, this shoud've failed correct?
EORI is country code followed by 12 or 15 digits.

File: mail/libraries/lite_to_edifact_converter.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MattHolmes123) [reply mail/libraries/lite_to_edifact_converter.py:326]: The next app type I am working on has several quantity codes so I will remove the hardcoded `030` `Q` and `O` values then

File: mail/icms_serializers.py [hunk 3]
```diff
@@ -61,14 +61,46 @@ class IcmsFaLicenceDataBaseSerializer(serializers.Serializer):
...
 
 
 class FaDflLicenceDataSerializer(IcmsFaLicenceDataBaseSerializer):
     """FA-DFL licence data serializer."""
 
-    country_code = serializers.CharField(max_length=2)
+    type = serializers.ChoiceField(choices=[LicenceTypeEnum.IMPORT_DFL])
+
+
+class FaSilGoods(GoodSerializer):
```
[the remark this answers is not in the document]
Comment (MattHolmes123) [reply mail/icms_serializers.py:89]: Updated.

File: mail/tests/test_serializers.py (hunk 2 above)
Comment (saruniitr) [reply mail/tests/test_serializers.py:421]: correct, max length allowed is 17 characters but actually it is country code followed by 12 or 15 digits only.

Comment (MattHolmes123) [reply mail/tests/test_serializers.py:421]: Sure but like I said I have a ticket to revisit this and will update this serializer when I get to `ICMSLST-1658`.
A test supplying 12 digits seems fine for now.
That's what the live ICMS records send on the trader line (12 digits)

**Review by saruniitr — APPROVED** [review]

**Pull request merged**